modules/auxiliary/scanner/http/iis_shortname_scanner.rb

```diff
@@ -0,0 +1,280 @@
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+  include Rex::Proto::Http
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'        => 'Microsoft IIS shortname vulnerability scanner',
+        'Description' => %q{
+        'The vulnerability is caused by a tilde character "~" in a GET or OPTIONS request, which
+         could allow remote attackers to diclose 8.3 filenames (short names). This is a new
+         technique for an existing bug (CVE-2009-4444), which relied on the use of the ;(semicolon)
+         character. Soroush Dalili discovered the original bug, while the new ~ technique was
+         discovered by Soroush Dalili and Ali Abbasnejad. Older IIS installations are vulnerable
+         with GET, newer installations with OPTIONS'
+        },
+        'Author'         =>
+          [
+          'MinatoTW <shaks19jais[at]gmail.com>',
+          'egre55 <ianaustin[at]protonmail.com>'
+          ],
+        'License'     => MSF_LICENSE,
+        'References'     =>
+          [
+            [ 'CVE', '2009-4444' ],
+            [ 'URL', 'https://soroush.secproject.com/blog/tag/iis-tilde-vulnerability/' ]
+          ],
+        'Targets' => [[ 'Automatic', {} ]]
+      )
+    )
+
+  register_options([
+          Opt::RPORT(80),
+          OptString.new('PATH', [ true, "The base path to start scanning from", "/" ])
+
+  ])
+  @dirs = []
+  @files = []
+  @threads = []
+  @queue = Queue.new
+  @queue_ext = Queue.new
+  @alpha = 'abcdefghijklmnopqrstuvwxyz0123456789!#$%&\'()-@^_`{}'
+  @found = Array.new
+  @found2 = Array.new
+  @found3 = Array.new
+  @verb = ""
+  end
+
+  def check
+    begin
+      for method in ['GET', 'OPTIONS']
+        res1 = send_request_cgi({
+          'uri' => normalize_uri(datastore['PATH'], '*~1*'),
+          'method' => method
+        })
+
+        res2 = send_request_cgi({
+          'uri' => normalize_uri(datastore['PATH'],'QYKWO*~1*'),
+          'method' => method
+        })
+
+        if res1.code == 404 && res2.code != 404
+          vuln = 1
+        end
+      end
+      if vuln == 1
+        return Exploit::CheckCode::Vulnerable
+      else
+        return Exploit::CheckCode::Safe
+      end
+
+      rescue Rex::ConnectionError
+        vprint_error("Connection failed")
+        return Exploit::CheckCode::Unknown
+    end
+  end
+
+  def is_vul
+      for method in ['GET', 'OPTIONS']
+        res1 = send_request_cgi({
+          'uri' => normalize_uri(datastore['PATH'], '*~1*'),
+          'method' => method
+        })
+
+        res2 = send_request_cgi({
+          'uri' => normalize_uri(datastore['PATH'],'QYKWO*~1*'),
+          'method' => method
+        })
+
+        if res1.code == 404 && res2.code != 404
+          vuln = 1
+          @verb = method
+          break
```

Comment:
As this branch calls `break`, and the remainder of the method does nothing except return `true` / `false` depending on comparison of the temporary variable `vuln == 1`, you could simply `return true` here, and `return false` at the end of the method. This way there would be no need to make use of the `vuln` temporary variable.